### DISCUSSION TITLE
ReferenceError: DateTime is not defined

### DISCUSSION THREAD

**Discussion by 25Das** [adonisjs/core#4639] [Help] [answered]
I am having trouble with adonisjs build process.

For some reason it doesn't import DateTime from Luxon.

In most models DateTime import is completely missing.

So when I run built server it crashs on seeders.

```
$ node build/bin/server.js
❯ error     database/seeders/main/index
  DateTime is not defined

   ReferenceError: DateTime is not defined

   at anonymous app\models\user.js:49
   44|      column({ serializeAs: null }),
   45|      __metadata("design:type", String)
   46|  ], User.prototype, "password", void 0);
   47|  __decorate([
   48|      column.dateTime({ autoCreate: true }),
 > 49|      __metadata("design:type", DateTime)
   50|  ], User.prototype, "createdAt", void 0);
   51|  __decorate([
   52|      column.dateTime({ autoCreate: true, autoUpdate: true }),
   53|      __metadata("design:type", DateTime)
   54|  ], User.prototype, "updatedAt", void 0);
   ```
   
   I made it work manually importing `import { DateTime } from 'luxon';` on every model missing DateTime import.
   
   Which was 13 of 15, 1 does not use DateTime and the other have the import from build, all the others haven't.
   
   All my seeders using this models broke too.
   
   This are my dependencies
   
   ```
   {
"devDependencies": {
  "@adonisjs/assembler": "^7.7.0",
  "@adonisjs/eslint-config": "^1.3.0",
  "@adonisjs/prettier-config": "^1.3.0",
  "@adonisjs/tsconfig": "^1.3.0",
  "@swc/core": "^1.6.6",
  "@types/luxon": "^3.4.2",
  "@types/node": "^20.14.9",
  "eslint": "^8.57.0",
  "hot-hook": "^0.2.6",
  "prettier": "^3.3.2",
  "ts-node": "^10.9.2",
  "typescript": "^5.5.2"
 },
 "dependencies": {
  "@adonisjs/auth": "^9.2.3",
  "@adonisjs/core": "^6.12.1",
  "@adonisjs/cors": "^2.2.1",
  "@adonisjs/lucid": "^21.1.0",
  "@vinejs/vine": "^2.1.0",
  "adonis-lucid-soft-deletes": "^2.0.0",
  "execa": "^9.3.0",
  "luxon": "^3.4.4",
  "mysql2": "^3.10.2",
  "pino-pretty": "^11.2.1",
  "reflect-metadata": "^0.2.2"
 },
}
```

**Comment by RomainLanz** [comment, marked as answer]
Hey @25Das! 👋🏻 

You are using a version of TypeScript that does not work with AdonisJS. Please downgrade to TypeScript 5.4.

We tweeted and informed people on our Discord server about this issue.

You can follow this issue to know when it's going to be fixed: https://github.com/microsoft/TypeScript/issues/59051

**Reply by 25Das** [reply]
Hi, @RomainLanz!

Sorry I dont use twitter neither discord.

Strangely I upgraded it today with adonisjs lastest versions and built with this error.

So I suspected and downgraded everything then built again and get the same error.

After your response I ensure downgrade of all deps I changed and rebuilt node_modules, run build again but the error remains.

I will track on ts issue you provided and on twitter for this problem.

Current packge.json deps

```
{
  "devDependencies": {
  "@adonisjs/assembler": "^7.7.0",
  "@adonisjs/eslint-config": "^1.3.0",
  "@adonisjs/prettier-config": "^1.3.0",
  "@adonisjs/tsconfig": "^1.3.0",
  "@swc/core": "^1.6.3",
  "@types/luxon": "^3.4.2",
  "@types/node": "^20.14.5",
  "eslint": "^8.57.0",
  "hot-hook": "^0.2.6",
  "prettier": "^3.3.2",
  "ts-node": "^10.9.2",
  "typescript": "^5.4.5"
 },
 "dependencies": {
  "@adonisjs/auth": "^9.2.3",
  "@adonisjs/core": "^6.12.0",
  "@adonisjs/cors": "^2.2.1",
  "@adonisjs/lucid": "^21.1.0",
  "@vinejs/vine": "^2.1.0",
  "adonis-lucid-soft-deletes": "^2.0.0",
  "execa": "^9.2.0",
  "luxon": "^3.4.4",
  "mysql2": "^3.10.1",
  "pino-pretty": "^11.2.1",
  "reflect-metadata": "^0.2.2"
 },
}
```

**Reply by RomainLanz** [reply]
Be careful; the version `^5.4.5` means everything in the `5.X.X` range.
You must use `~5.4` to stay in `5.4.X`.

**Reply by 25Das** [reply]
You are totally right, I always thought if patch version number is set then '^' would held dependency to the minor version number not the major.

Well problem solved! Thanks @RomainLanz
